molgenis/molgenis-emx2
# fix(directory): wrong number in number of samples in fact table when at least one Split by is disabled. closes #3969

closes #3969

What are the main changes you did:
- explain what you changed and essential considerations.
I added the case when collapsedFact[column]=fact[column] that happens frequently in number of samples column where 1 is the most frequent value.

how to test:
- explain here what to do to test this (or point to unit tests)
I suggest to use what I described in the 3969 issue, that is the urine-male-middle-aged-all count, but any other count can be used. The count with "Split by" are correct so if the count differ with the one obtained when at least one column where "Split by" is disabled then the error still exists.
I'm gonna report the one I chose for the issue creation below for convenience:
Steps to reproduce the behavior when the issue is present:

    Go to 'https://directory-emx2-acc.molgenis.net/ERIC/directory/#/collection/bbmri-eric:ID:EXT_76823:collection:MainCollection?search=76823'
    (All Split by are initially enabled)
    Select Urine from first column, Male for second one, Middle-aged for third one
    Sum manually the samples. That is 3. This is correct and consistent with the saved fact table csv file.
    Now Uncheck "Split" by for "Disease codes" column.
    Select again Urine- Male- Middle-aged. The number of samples yielded is 1. It should be 3.


todo:
